iothub/device/src/Logger/EventCounterLogger.cs

```diff
@@ -0,0 +1,347 @@
+                EventCounter amqpSessionEstablishEventCounter = new EventCounter("AMQP-Session-Establish", this);
+                EventCounter amqpSessionDisconnectionEventCounter = new EventCounter("AMQP-Session-Disconnection", this);
+                EventCounter amqpTokenRefreshStartEventCounter = new EventCounter("AMQP-Token-Refresh-Start", this);
+                EventCounter amqpTokenRefreshStopEventCounter = new EventCounter("AMQP-Token-Refresh-Stop", this);
+                EventCounter amqpTokenRefreshEventCounter = new EventCounter("AMQP-Token-Refresh", this);
+
+                while (!cancellationToken.IsCancellationRequested)
+                {
+                    deviceClientCreationEventCounter.WriteMetric(_deviceClientCreationCounts);
+                    deviceClientDisposeEventCounter.WriteMetric(_deviceClientDisposeCounts);
+                    amqpUnitCreationEventCounter.WriteMetric(_amqpUnitCreationCounts);
+                    amqpUnitDisposeEventCounter.WriteMetric(_amqpUnitDisposeCounts);
+                    amqpConnectionEstablishEventCounter.WriteMetric(_amqpConnectionEstablishCounts);
+                    amqpConnectionDisconnectionEventCounter.WriteMetric(_amqpConnectionDisconnectionCounts);
+                    amqpSessionEstablishEventCounter.WriteMetric(_amqpSessionEstablishCounts);
+                    amqpSessionDisconnectionEventCounter.WriteMetric(_amqpSessionDisconnectionCounts);
+                    amqpTokenRefreshStartEventCounter.WriteMetric(_amqpTokenRefreshStartCounts);
+                    amqpTokenRefreshStopEventCounter.WriteMetric(_amqpTokenRefreshStopCounts);
+                    amqpTokenRefreshEventCounter.WriteMetric(_amqpTokenRefreshCounts);
+                    await Task.Delay(interval).ConfigureAwait(false);
+                }
+            }
+            if (redirectToConsole)
+            {
+                Console.WriteLine("Time,Device-Client-Creation,Device-Client-Dispose,AMQP-Unit-Creation,AMQP-Unit-Dispose,AMQP-Connection-Establish,AMQP-Connection-Disconnection,AMQP-Session-Establish,AMQP-Session-Disconnection,AMQP-Token-Refresher-Started,AMQP-Token-Refresher-Stopped,AMQP-Token-Refreshes");
+                while (!cancellationToken.IsCancellationRequested)
+                {
+                    Console.WriteLine($"{DateTime.Now},{_deviceClientCreationCounts},{_deviceClientDisposeCounts},{_amqpUnitCreationCounts},{_amqpUnitDisposeCounts},{_amqpConnectionEstablishCounts},{_amqpConnectionDisconnectionCounts},{_amqpSessionEstablishCounts},{_amqpSessionDisconnectionCounts},{_amqpTokenRefreshStartCounts},{_amqpTokenRefreshStopCounts},{_amqpTokenRefreshCounts}");
+                    await Task.Delay(interval).ConfigureAwait(false);
+               }
+            }
+
+            _started = false;
+        }
+    }
+# else
+    internal class EventCounterLogger : IEventCounterLogger
```

Comment:
Nit: this doesn't appear to be available in Net451. Let's chat offline - maybe we can implement it there with `System.Diagnostics.PerformanceCounter`?